src/sql/platform/query/common/gridDataProvider.ts

```diff
@@ -23,7 +23,7 @@ export interface IGridDataProvider {
 	 * @param resultId The result id of the result to copy from
 	 * @param includeHeaders [Optional]: Should column headers be included in the copy selection
 	 */
-	copyResults(selection: Slick.Range[], includeHeaders?: boolean): void;
+	copyResults(selection: Slick.Range[], includeHeaders?: boolean): Promise<void>;
```

Comment:
want to add `async` here?

Reply:
Don't think you can for method definitions.

Reply:
async is more of a functional keyword than a typing keyword, so having something like 
`async function: Promise<void>` would be unnecessary.

Reply:
ah i didn't see it was a method def